addon/-private/data-view/radar/radar.js

```diff
@@ -530,11 +531,19 @@ export default class Radar {
     if (_componentPool.length > 0) {
       if (shouldRecycle === true) {
         // Grab the DOM of the remaining components and move it to temporary node disconnected from
-        // the body. If we end up using these components again, we'll grab their DOM and put it back
-        for (let i = 0; i < _componentPool.length; i++) {
+        // the body if the item can be reused later otherwise delete the component to avoid virtual re-rendering of the
+        // deleted item. If we end up using these components again, we'll grab their DOM and put it back
+        for (let i = _componentPool.length - 1; i >= 0; i--) {
           const component = _componentPool[i];
-
-          insertRangeBefore(this._domPool, null, component.realUpperBound, component.realLowerBound);
+          const item = objectAt(items, component.index);
+          if (item) {
+            insertRangeBefore(this._domPool, null, component.realUpperBound, component.realLowerBound);
+          } else {
+            run(() => {
+              virtualComponents.removeObject(component);
+            });
+            _componentPool.splice(i, 1);
```

Comment:
Ah, the reverse traversal is so this mutation is safe, yes?

The `insertRangeBefore(` logic doesn't care about the order of traversal? I take it the answer is no, but I'm not so familiar with the code that I want to presume it without diving in.

Reply:
yep, as the indexes are only decreasing you are sure you don't miss any item of the _componentPool in the loop.
There is no sense in order in `insertRangeBefore` as the `_componentPool` is shuffled anyway